src/Kobo/PowerOff.cpp

```diff
@@ -84,9 +85,18 @@ DrawBanner(Canvas &canvas, PixelRect &rc)
   canvas.DrawText({x, rc.top + int(banner_height - normal_font.GetHeight()) / 2},
                   website);
 
-  const TCHAR *const comment = _T("powered off");
-  canvas.DrawText({rc.right - (int)canvas.CalcTextWidth(comment) - padding, rc.top + padding},
-                  comment);
+  TCHAR battery[4];
+  if (File::ReadString(Path("/sys/class/power_supply/mc13892_bat/capacity"), battery, sizeof(battery))){
+          TCHAR comment[] = _T("powered off - battery ");
+          strcat(comment, battery);
+          strcat(comment, "%");
```

Comment:
Stack buffer overflow bug here, may crash.

Reply:
Defining a size for the variable "comment" should be enough, right? Like:
`TCHAR comment[30];`

Reply:
If you believe that's enough, then yes - and please use `snprintf()` instead of `strcat()` - the latter is a bad API.